src/Action/InitElasticSearch.php

```diff
@@ -125,10 +122,21 @@ public function __invoke($params): Report
         $config[GenerisSearch::class] = new GenerisSearch();
 
         try {
-            $search = new ElasticSearch($config);
-
-            $search->createIndexes();
-            $this->getServiceManager()->register(Search::SERVICE_ID, $search);
+            $elasticSearch = new ElasticSearch($config);
+            $elasticSearch->createIndexes();
+
+            /** @var SearchProxy $search */
+            $search = $this->getServiceManager()->get(SearchProxy::SERVICE_ID);
+            
+            if ($search instanceof GenerisSearch) {
+                $this->getServiceManager()->register(Search::SERVICE_ID, $elasticSearch);
+            }
+            
+            if ($search instanceof SearchProxy) {
+                $search->setOption(SearchProxy::OPTION_ADVANCED_SEARCH_CLASS, $elasticSearch);
```

Comment:
Just a note to update tao-core version on composer before merge

 - [ ] Update tao-core